packages/clay-card/src/Body.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import React from 'react';
+
+import classNames from 'classnames';
+
+import Context from './Context';
+
+export interface CardBodyProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+export const Body: React.FunctionComponent<CardBodyProps> = ({
+	children,
+	className,
+	...otherProps
+}) => {
+	const {horizontal, interactive} = React.useContext(Context);
+
+	const TagName = interactive ? 'span' : 'div';
+
+	const Content: React.FunctionComponent<CardBodyProps> = ({children}) => (
```

Comment:
Instead of creating a function here, I think you can do something like

```jsx
	const content = (
		<TagName className={classNames('card-body', className)} {...otherProps}>
			{interactive && !horizontal ? (
				children
			) : (
				<TagName className="card-row">{children}</TagName>
			)}
		</TagName>
	);

	if (horizontal && !interactive) {
		return <div className="card card-horizontal">{content}</div>;
	}

	return content;
```